tests/machine_control/test_machine_controller.py

```diff
@@ -2171,97 +2236,114 @@ def test_get_num_working_cores(self, num_cpus):
         cn.read_struct_field.return_value = num_cpus
 
         assert cn.get_num_working_cores(x=0, y=0) == num_cpus
-        assert cn.read_struct_field.called_once_with("sv", "num_cpus", 0, 0, 0)
+        cn.read_struct_field.assert_called_once_with("sv", "num_cpus", 0, 0)
+
+    @pytest.mark.parametrize("arg1,num_cores,working_links",
+                             [((18 | (0b111110 << 8)),
+                               18,
+                               set(l for l in Links if l != Links.east)),
+                              ((18 | (0b011111 << 8)),
+                               18,
+                               set(l for l in Links if l != Links.south)),
+                              ((17 | (0b111111 << 8)),
+                               17,
+                               set(Links)),
+                              ])
```

Comment:
Sadly even doing that does not allow the stimulus lines to fit on one line so it probably isn't worth changing in this case. In general though I'd agree!